server/middleware/mentor/index.js

```diff
@@ -12,41 +13,164 @@ const resObj = (statusCode, message, data = {}) => ({
   data,
 });
 
-const queryMentors = async ({ email, role_id }) => {
+const createPayload = (body, mentor_id) => {
+  const { bio, skills, phone, timezone, googlemeet, timeSlot } = body;
+  return {
+    ...(mentor_id ? { mentor_id: mentor_id } : {}),
+    ...(bio ? { bio: bio } : {}),
+    ...(skills ? { skills: skills } : {}),
+    ...(phone ? { phone: phone } : {}),
+    ...(timezone ? { timezone: timezone } : {}),
+    ...(googlemeet ? { googlemeet: googlemeet } : {}),
+    ...(timeSlot ? { timeSlot: timeSlot } : {}),
+  };
+};
+
+const saveMentorInfo = async (req, res, tokenData) => {
   let data;
   let statusCode;
   let message;
-  let collectionObj;
+  const { USERS_COLLECTION } = process.env;
+  const { MENTORS_COLLECTION } = process.env;
 
-  logger.info(
-    setLogDetails(
-      'mentorMiddleware.getMentors',
-      'Fetching metors data',
-      `User - ${email}`
-    )
-  );
+  try {
+    const mentorPayload = createPayload(req.body, tokenData.user_id);
+    const mentorCollection = await mongoUtil.fetchCollection(
+      MENTORS_COLLECTION
+    );
+    const saveInfo = await mentorCollection.insertOne(mentorPayload);
+    const { password, ...rest } = saveInfo.ops[0];
+    data = rest;
+    if (!data) {
+      throw Error('Something went wrong!');
+    } else {
+      statusCode = OK;
+      message = 'Success';
+    }
+    const userCollection = await mongoUtil.fetchCollection(USERS_COLLECTION);
+    const updateUser = await userCollection.findOneAndUpdate(
+      { email: tokenData.email },
+      { $set: { hasOnboarded: true } },
+      { returnOriginal: false }
+    );
+    const { pass, ...value } = updateUser.value;
+    generateCookie(res, value);
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
```

Comment:
log in event of error

Reply:
resolve in commit #https://github.com/femmecubator/femmecubator-api-repo/pull/72/commits/58b6dd83fe0bda2e2c2dc2899c8593f45bc7bcc0